src/ui/components/search/searchcomponent.js

```diff
@@ -344,59 +374,96 @@ export default class SearchComponent extends Component {
     }
   }
 
-  requestIconAnimationFrame (iconState) {
+  requestIconAnimationFrame (iconState, iconElClass) {
```

Comment:
With the added complexity of of the new animated icon, I think it makes sense to consider creating a new class that manages the styling of the icon component. I don't think it needs to be a full-blown component. It could be called the IconAnimator, and it could provide a function to accept a state, and in response it would toggle the corresponding CSS classes. It would allow us to remove icon-specific styling from the search component class

Reply:
Is it possible to remove `iconElClass` from this param and instead derive it from the iconState?

Reply:
The classnames of the icon components to get the right styling in the logic would still be tied to the searchcomponent so it wouldn't be general enough for something else to reuse the function but I could see that having a separate class may make it more organized.
right now, LOADING state could refer default or custom so there's a need for `iconElClass` .but If I add `CUSTOM_LOADING` to the IconState and adjust some of the logic, then yeah we can just derive it from iconState!